src/PAModel/Serializers/MsAppSerializer.cs

```diff
@@ -299,16 +299,18 @@ public static CanvasDocument Load(Stream streamToMsapp, ErrorContainer errors)
                     app._properties.LocalConnectionReferences = null;
                 }
 
-                if (!string.IsNullOrEmpty(app._properties.LocalDatabaseReferences))
+                IDictionary<String, LocalDatabaseReferenceJson> dsrs = null;
+                if (!string.IsNullOrEmpty(app._properties.LocalDatabaseReferences) && (dsrs = Utilities.JsonParse<IDictionary<String, LocalDatabaseReferenceJson>>(app._properties.LocalDatabaseReferences)).Count > 0)
```

Comment:
Assignment that can be short circuited is a little hard to read. Could this be refactored to not have the assignment part of the condition?

Reply:
Refactored this block to remove assignment from the condition. Need to check for empty string or null before performing Json parsing.